### PR TITLE
Rename QMCAppBase and QMCMain members and one ifdef QMC_COMPLEX

### PR DESCRIPTION
**Pull request by williamfgc**
## Proposed changes

Rename protected and private members
Eliminate one #ifdef QMC_COMPLEX in QMCMain class
Clean unused member in QMCMain
Related to #3411 #4099

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
Ubuntu 22, must pass CI variants

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Review by markdewing — DISMISSED** [review]

**Review by ye-luo — APPROVED** [review]

**Review by prckent — DISMISSED** [review]
LGTM

(One QMC_COMPLEX removed, 665 to go.)

**Comment by williamfgc** [comment]
CI reveals the `private: hdf_error_suppression hide_hdf_errors_;`  is not very private :)

**Comment by williamfgc** [comment]
Test this please

**Comment by williamfgc** [comment]
Test this please

File: src/QMCTools/QMCFiniteSize/QMCFiniteSize.cpp [hunk 1]
```diff
@@ -79,16 +79,15 @@ bool QMCFiniteSize::validateXML()
...
         popDocument();
       }
     }
     else if (cname == "qmcsystem")
     {
       processPWH(cur);
     }
-    else
-    {}
+    else {}
```
Comment (ye-luo) [inline src/QMCTools/QMCFiniteSize/QMCFiniteSize.cpp:90]: I think we can simply remove this line. Please take care of this in your next PR.

**Review by ye-luo — APPROVED** [review]

**Pull request merged**